subprojects/packagefiles/libscpi/libscpi/meson.build

```diff
@@ -0,0 +1,29 @@
+subdir('src')
+
+libscpi_lib = both_libraries('scpi',
```

Comment:
Why not `library`? (And let the user decide with `--default-library=…`)

Reply:
For the tests?

Reply:
Yes, for the tests. When I tried to use a shared library, the tests could not compile because they refer to internal symbols that are not exported in the SO

Reply:
Is there a portable way to say "export all symbols?"

Reply:
How are symbols exported? You're not passing any specific arguments for building a static or shared variant.

Reply:
Right, `__attribute__((visibility ("hidden")))` is hiding some symbols needed for the tests.

What I would do: always build an internal static library (`libscpi_static`), and use it for the tests and to build the exported library (using `library(…, link_with: libscpi_static, …)`:

```diff
--- a/libscpi/meson.build
+++ b/libscpi/meson.build
@@ -1,9 +1,19 @@
 subdir('src')
 
-libscpi_lib = both_libraries('scpi',
+libscpi_inc = include_directories('inc')
+
+libscpi_static = static_library(
+  'scpi_static',
   libscpi_srcs,
-  include_directories : 'inc',
-  install : true,
+  include_directories: libscpi_inc,
+)
+
+libscpi_lib = library(
+  'scpi',
+  libscpi_srcs,
+  link_with: libscpi_static,
+  include_directories: libscpi_inc,
+  install: true,
 )
 
 libscpi_dep = declare_dependency(
@@ -23,7 +33,6 @@
   description : 'Open Source SCPI device library',
   subdirs : 'scpi',
   libraries : libscpi_lib,
-  version : '2.3',
 )
 
 subdir('test')
--- a/libscpi/test/meson.build
+++ b/libscpi/test/meson.build
@@ -12,13 +12,13 @@
   'test_scpi_utils',
 ]
 
-libscpi_static = libscpi_lib.get_static_lib()
+# libscpi_static = libscpi_lib.get_static_lib()
 
 foreach t : test_names
   test_exe = executable(t, t+'.c',
     dependencies : [ cunit_dep ],
     link_with : libscpi_static,
-    include_directories : '../inc',
+    include_directories : libscpi_inc,
   )
   test(t, test_exe)
 endforeach
```

Reply:
there was a different way to do this. export_all_objects or something.

Reply:
https://mesonbuild.com/Reference-manual_returned_build_tgt.html#build_tgtextract_all_objects

Reply:
Yep, forgot about that.

Reply:
Yes, that works great, added.